marquez_client/marquez.py

```diff
@@ -100,7 +100,7 @@ def get_all_datasources(self, limit=None, offset=None, **kwargs):
 
     @handle_response
     def get_datasource(self, urn, **kwargs):
-        path = "datasources/{0}".format(urllib.parse.quote(str(urn), safe=':'))
+        path = "datasources/{0}".format(urllib.parse.quote(str(urn)))
```

Comment:
I would expect URL-encoding to happen at a much lower level so that all API methods get it for free

Reply:
think about it as a `BASE_URL` with an optional number of path "parts", e.g.:

`http://random-url.com/api/v1/jobs/foo/bar` is broken up as 

```
{
    'base_url': 'http://random-url.com/api/v1/jobs',
    'parts': ['foo', 'bar']
}
```

If you url-encode each part before you build up the URL, you'll be safe

Reply:
Yup, that was the idea -- I url_encode the request args much closer to where they're used by `requests`. For url-encoding the path, I think I can move the url-encoding closer to where it's used (i.e. lower level), but what was tripping me up was what would happen if there are forward slashes in the job name. Which is why I wanted to sanitize each part of the path individually.

That said, there is likely a way to do it with that in mind, so I'll need to think about that and add it. I'll add some test cases too.

Reply:
cool 👍 looking forward to seeing this pushed down

Reply:
Current version of the code has this fix now. It's been pushed down with the encoding work removed from the client portion. It's also a nice separation of concerns, as the client portion should be more concerned with the 'what' and less of the 'how'.